src/client/real.rs

```diff
@@ -296,13 +311,42 @@ impl Client {
                     .extend(scope.tags.iter().map(|(k, v)| ((*k).clone(), (*v).clone())));
             }
 
-            if !scope.contexts.is_empty() {
-                event.contexts.extend(
-                    scope
-                        .contexts
-                        .iter()
-                        .map(|(k, v)| ((*k).clone(), (*v).clone())),
-                );
+            {
+                let mut add_os = true;
+                let mut add_rust = true;
+                let mut add_device = true;
+
+                for (key, value) in scope.contexts.iter() {
+                    match *value {
+                        None => {
+                            match key.as_str() {
+                                "os" => add_os = false,
+                                "rust" => add_rust = false,
+                                "device" => add_device = false,
+                                _ => {},
+                            }
+                        }
+                        Some(ref value) => {
+                            event.contexts.insert((*key).clone(), (*value).clone());
```

Comment:
This should not override existing contexts in the event.

Reply:
Never mind, just saw that map does not override when calling `insert`. You could simplify this for the three other contexts then and just "attempt" an insert, instead of keeping three flags...

Reply:
We can change that. Current behavior was already to override.